includes/class-apple-news.php

```diff
@@ -53,4 +61,29 @@ public static function get_filename( $path ) {
 		// Remove any spaces and return the filename
 		return str_replace( ' ', '', $filename );
 	}
+
+	/**
+	 * Attempt to migrate settings from an older version of this plugin
+	 *
+	 * @param Settings $settings
+	 */
+	public function migrate_settings( $settings ) {
+		$migrated_settings = array();
+
+		// For each potential value, see if the WordPress option exists.
+		// If so, migrate its value into the new array format.
+		// If it doesn't exist, just use the default value.
+		foreach ( $settings->all() as $key => $default ) {
+			$value = get_option( $key, $default );
+			$migrated_settings[ $key ] = $value;
+
+			// Delete the option in case it existed to clean up
+			delete_option( $key );
+		}
+
+		// Store these settings
+		update_option( $this->option_name, $migrated_settings );
```

Comment:
I think you should set `$autoload` to no on these options, unless they need to be used enough on the front end to be worth always having in memory. This note applies everywhere `update_option` is called.